extension/datadogfleetautomationextension/transform.go

```diff
@@ -0,0 +1,353 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadogfleetautomationextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/datadogfleetautomationextension"
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/service"
+	"go.uber.org/zap"
+)
+
+const (
+	receiverKind   = "receiver"
+	receiversKind  = "receivers"
+	processorKind  = "processor"
+	processorsKind = "processors"
+	exporterKind   = "exporter"
+	exportersKind  = "exporters"
+	extensionKind  = "extension"
+	extensionsKind = "extensions"
+	connectorKind  = "connector"
+	connectorsKind = "connectors"
+	providerKind   = "provider"
+	providersKind  = "providers"
+	converterKind  = "converter"
+	convertersKind = "converters"
+	pipelinesKind  = "pipelines"
+)
+
+var kindsToKind = map[string]string{
+	receiversKind:  receiverKind,
+	processorsKind: processorKind,
+	exportersKind:  exporterKind,
+	extensionsKind: extensionKind,
+	connectorsKind: connectorKind,
+	providersKind:  providerKind,
+	convertersKind: converterKind,
+}
+
+type ComponentChecker interface {
+	isComponentConfigured(string, string) (bool, *component.ID)
+	isModuleAvailable(string, string) bool
+	isHealthCheckV2Enabled() (bool, error)
+}
+
+type defaultComponentChecker struct {
+	extension *fleetAutomationExtension
+}
+
+func (d *defaultComponentChecker) isComponentConfigured(typ string, componentsKind string) (bool, *component.ID) {
+	return d.extension.isComponentConfigured(typ, componentsKind)
+}
+
+func (d *defaultComponentChecker) isModuleAvailable(componentType string, componentKind string) bool {
+	return d.extension.isModuleAvailable(componentType, componentKind)
+}
+
+func (d *defaultComponentChecker) isHealthCheckV2Enabled() (bool, error) {
+	return d.extension.isHealthCheckV2Enabled()
+}
```

Comment:
can we decouple component checker and extension. may be make component checker its own module